apps/whale-api/src/module.api/poolpair.prices.service.ts

```diff
@@ -73,32 +64,34 @@ export class PoolPairPricesService {
   }
 
   /**
-   * Helper to get all tokens in a map indexed by their symbol for quick look-ups
+   * Helper to get all tokens from defid and filters them
    * @private
-   * @return {Promise<TokensBySymbol>}
+   * @return {Promise<TokenInfoWithId[]>}
    */
-  private async getAllTokens (): Promise<TokensBySymbol> {
-    const tokens = await this.tokenMapper.query(Number.MAX_SAFE_INTEGER)
-    const allTokenInfo: TokensBySymbol = {}
+  private async getAllTokens (): Promise<TokenInfoWithId[]> {
+    const tokens = await this.defidCache.getAllTokenInfo() ?? []
+    return tokens.filter(token => !this.isUntradeableToken(token))
+  }
 
-    for (const token of tokens) {
-      // Skip LP tokens and non-DAT tokens
-      if (token.isLPS || !token.isDAT || token.symbol === 'BURN') {
-        continue
-      }
+  private async getTokenBySymbol (symbol: string): Promise<TokenInfoWithId> {
+    let tokenResult: TokenResult | undefined
 
-      const tokenInfo = await this.defidCache.getTokenInfo(token.tokenId.toString())
-      if (tokenInfo === undefined) {
-        this.logger.error(`Could not find token info for id: ${token.tokenId}`)
-        continue
-      }
+    try {
+      tokenResult = await this.defidCache.getTokenInfoBySymbol(symbol)
+    } catch (e) {
+      throw new TokenSymbolNotFoundError(symbol)
+    }
 
-      allTokenInfo[token.symbol] = {
-        id: token.tokenId.toString(),
-        ...tokenInfo
-      }
+    if (tokenResult === undefined || Object.keys(tokenResult).length === 0) {
+      throw new TokenSymbolNotFoundError(symbol)
     }
-    return allTokenInfo
+
+    const [id, tokenInfo] = Object.entries(tokenResult)[0]
+    return { ...tokenInfo, id }
+  }
+
+  private isUntradeableToken (token: TokenInfoWithId): boolean {
+    return token.isLPS || !token.isDAT || token.symbol === 'BURN'
```

Comment:
hmm... not sure want to hide old token(eg: AMZN-DUSD/v1) as well?? :thinking: 
https://ocean.defichain.com/v0/mainnet/poolpairs/54